stl/inc/deque

```diff
@@ -557,9 +555,21 @@ public:
 
     _Deque_val() noexcept : _Map(), _Mapsize(0), _Myoff(0), _Mysize(0) {}
 
-    size_type _Getblock(size_type _Off) const noexcept {
+    _Map_difference_type _Getblock(size_type _Off) const noexcept {
         // NB: _Mapsize and _Block_size are guaranteed to be powers of 2
-        return (_Off / _Block_size) & (_Mapsize - 1);
+        return static_cast<_Map_difference_type>((_Off / _Block_size) & (_Mapsize - 1));

```

Comment:
I think we should put off every `cast-to-_Map_difference_type` to caller site, *only when having to add to a pointer*.